letshelp-letsencrypt/letshelp_letsencrypt/letshelp_letsencrypt_apache.py

```diff
@@ -0,0 +1,303 @@
+        # don't include it in selection. It is safe to include the file if
+        # it consists solely of comments because comments are removed before
+        # submission.
+        return empty_or_all_comments or not possible_password_file
+
+    return False
+
+
+def setup_tempdir(args):
+    """Creates a temporary directory and necessary files for config
+
+    :param argparse.Namespace args: Parsed command line arguments
+
+    :returns: Path to temporary directory
+    :rtype: str
+
+    """
+    tempdir = tempfile.mkdtemp()
+
+    with open(os.path.join(tempdir, "config_file"), "w") as config_fd:
+        config_fd.write(args.config_file + "\n")
+
+    proc = subprocess.Popen([args.apache_ctl, "-v"],
+                            stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    with open(os.path.join(tempdir, "version"), "w") as version_fd:
+        version_fd.write(proc.communicate()[0])
+
+    proc = subprocess.Popen([args.apache_ctl, "-d", args.server_root, "-f",
+                             args.config_file, "-M"],
+                            stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    with open(os.path.join(tempdir, "modules"), "w") as modules_fd:
+        modules_fd.write(proc.communicate()[0])
+
+    proc = subprocess.Popen([args.apache_ctl, "-d", args.server_root, "-f",
+                             args.config_file, "-t", "-D", "DUMP_VHOSTS"],
+                            stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    with open(os.path.join(tempdir, "vhosts"), "w") as vhosts_fd:
+        vhosts_fd.write(proc.communicate()[0])
+
+    return tempdir
+
+
+def verify_config(args):
+    """Verifies server_root and config_file specify a valid config
+
+    :param argparse.Namespace args: Parsed command line arguments
+
+    """
+    try:
+        with open(os.devnull, "w") as devnull:
+            subprocess.check_call([args.apache_ctl, "-d", args.server_root,
+                                   "-f", args.config_file, "-t"],
+                                  stdout=devnull, stderr=subprocess.STDOUT)
+    except OSError:
+        sys.exit(_NO_APACHECTL)
+    except subprocess.CalledProcessError:
+        sys.exit("Syntax check from apachectl failed")
+
+
+def locate_config(apache_ctl):
+    """Uses the apachectl binary to find configuration files
+
+    :param str apache_ctl: Path to `apachectl` binary
+
+
+    :returns: Path to Apache server root and main configuration file
+    :rtype: `tuple` of `str`
+
+    """
+    try:
+        proc = subprocess.Popen([apache_ctl, "-V"],
+                                stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        output, _ = proc.communicate()
+    except OSError:
+        sys.exit(_NO_APACHECTL)
+
+    server_root = config_file = ""
+    for line in output.splitlines():
+        # Relevant output lines are of the form: -D DIRECTIVE="VALUE"
+        if "HTTPD_ROOT" in line:
+            server_root = line[line.find("\"")+1:-1]
```

Comment:
IMO `'"'` instead of backslash escape would be more readable and is in line with rest of the code base, here and below